src/EditorFeatures/Core/ExtractMethod/ExtractMethodCommandHandler.cs

```diff
@@ -210,14 +216,14 @@ public bool ExecuteCommand(ExtractMethodCommandArgs args, CommandExecutionContex
         }
 
         private void ApplyChange_OnUIThread(
-            ITextBuffer textBuffer, IEnumerable<TextChange> changes, IBackgroundWorkIndicatorContext waitContext)
+            ITextBuffer textBuffer, IEnumerable<TextChange> changes, IBackgroundWorkIndicator workIndicator)
         {
             _threadingContext.ThrowIfNotOnUIThread();
 
             using var undoTransaction = _undoManager.GetTextBufferUndoManager(textBuffer).TextBufferUndoHistory.CreateTransaction("Extract Method");
 
             // We're about to make an edit ourselves.  so disable the cancellation that happens on editing.
-            waitContext.CancelOnEdit = false;
+            using var _ = workIndicator.SuppressAutoCancel();
```

Comment:
nice.  i like thsi pattern.

Reply:
Are we sure this method applies to cancel on edit? There is also a feature to automatically cancel operations which take too much time and it's not clear from the method name which one is intended.

Reply:
This applies to the BackgroundWorkIndicatorOptions set on the indicator